adam/perception/perception_graph.py

```diff
@@ -313,111 +328,113 @@ def from_schema(object_schema: ObjectStructuralSchema) -> "PerceptionGraphPatter
         """
         Creates a pattern for recognizing an object based on its *object_schema*.
         """
-        graph = DiGraph()
-        object_to_node: Dict[Any, NodePredicate] = {}
-
-        PerceptionGraphPattern._translate_schema(
-            # since the root object does not itself have a `SubObject` object,
-            # we wrap it in one.
-            object_=SubObject(object_schema),
-            object_schema=object_schema,
-            graph=graph,
-            schema_node_to_pattern_node=object_to_node,
+
+        # Create a PerceptionGraph from schema
+        schema_situation_object = SituationObject.instantiate_ontology_node(
+            ontology_node=object_schema.ontology_node, ontology=GAILA_PHASE_1_ONTOLOGY
+        )
+        situation = HighLevelSemanticsSituation(
+            ontology=GAILA_PHASE_1_ONTOLOGY, salient_objects=[schema_situation_object]
+        )
+        perception_generator = HighLevelSemanticsSituationToDevelopmentalPrimitivePerceptionGenerator(
+            GAILA_PHASE_1_ONTOLOGY
+        )
+        perception = perception_generator.generate_perception(
+            situation, chooser=RandomChooser.for_seed(0)
         )
-        return PerceptionGraphPattern(graph)
+        perception_graph = PerceptionGraph.from_frame(first(perception.frames))
+
+        # We are removing certain information that is added by situation but is not in schema
+        # We remove: colors, ground, learner, regions, properties
+        nodes_to_remove = [
+            node
+            for node in perception_graph.graph.nodes
+            if isinstance(node, RgbColorPerception)
+            or (
+                isinstance(node, ObjectPerception)
+                and node.debug_handle == GROUND_PERCEPTION.debug_handle
+            )
+            or (
+                isinstance(node, ObjectPerception)
+                and node.debug_handle == LEARNER_PERCEPTION.debug_handle
+            )
+            or (isinstance(node, tuple) and isinstance(node[0], Region))
+            or (isinstance(node, OntologyNode))
+        ]
+
+        perception_graph.graph.remove_nodes_from(nodes_to_remove)
```

Comment:
I think we want to keep `PerceptionGraph` as immutable, but you can get a copy of the underlying NetworkX `DiGraph`, modify it, and make a new `PerceptionGraph`